### PR TITLE
preserve VP9 parsing context during encryption

### PR DESCRIPTION
**Pull request by cconcolato**

### CODE REVIEW

**Comment by rbouqueau** [comment]
I made a comment. Otherwise LGTM on the VP9 side. I let @jeanlf check for the (de)allocation positions.

File: src/media_tools/av_parsers.c [hunk 1]
```diff
@@ -1933,9 +1933,10 @@ GF_Err gf_media_vp9_parse_superframe(GF_BitStream *bs, u64 ivf_frame_size, u32 *
 	*superframe_index_size = 2 + bytes_per_framesize * *num_frames_in_superframe;
 	gf_bs_seek(bs, pos + ivf_frame_size - *superframe_index_size);
 	byte = gf_bs_read_u8(bs);
-	if ((byte & 0xe0) != 0xc0)
+	if ((byte & 0xe0) != 0xc0) {
+		*num_frames_in_superframe = 0; // force failure
 		goto exit; /*no superframe*/
-
+	}
```
[the remark this answers is not in the document]
Comment (cconcolato) [reply src/media_tools/av_parsers.c:1939]: above, it is not an error case, and the value of num_frames_in_superframe has not been modified. Maybe we should revert it to 1 (and not 0?)

[the remark this answers is not in the document]
Comment (rbouqueau) [reply src/media_tools/av_parsers.c:1939]: After looking again, I think that if the sync markers have the wrong values we should return a ```GF_NON_COMPLIANT_BITSTREAM```. Then we can stop checking ```if (e || !num_frames_in_superframe) return e;```and look after ```e``` only. Error propagation is ok in the codebase.

CC @jeanlf

[the remark this answers is not in the document]
Comment (cconcolato) [reply src/media_tools/av_parsers.c:1939]: I pushed an additional commit that does that.

[the remark this answers is not in the document]
Comment (rbouqueau) [reply src/media_tools/av_parsers.c:1939]: Perfect. Merging.

**Pull request merged**